ragaai_catalyst/tracers/agentic_tracing/upload/session_manager.py

```
@@ -0,0 +1,156 @@
+            'User-Agent': 'RagaAI-Catalyst/1.0'
+        })
+
+        logger.info("HTTP session initialized successfully with adapters mounted for http:// and https://")
+
+        # Warm up connection pool using RagaAICatalyst.BASE_URL
+        if os.getenv("RAGAAI_CATALYST_BASE_URL") is not None:
+            base_url = os.getenv("RAGAAI_CATALYST_BASE_URL")
+            logger.info(f"Warming up connection pool using RagaAICatalyst.BASE_URL: {base_url}")
+            self.warm_up_connections(base_url)
+        else:
+            logger.warning(f"RAGAAI_CATALYST_BASE_URL not available, skipping connection warmup")
```

Flake8 (7.2.0)

[error] f-string is missing placeholders

(F541)